uaclient/log.py

```diff
@@ -9,3 +12,53 @@ class RedactionFilter(logging.Filter):
     def filter(self, record: logging.LogRecord):
         record.msg = util.redact_sensitive_logs(str(record.msg))
         return True
+
+
+class JsonArrayFormatter(logging.Formatter):
+    """Json Array Formatter for our logging mechanism
+    Custom made for Pro logging needs
+    """
+
+    default_time_format = "%Y-%m-%dT%H:%M:%S%z"
```

Comment:
Let's add microseconds here
```suggestion
    default_time_format = "%Y-%m-%dT%H:%M:%S.%f%z"
```

Reply:
`%f` does not work, but there is another way of using `record.msecs` with `record.asctime` which I have used.